### PR TITLE
[OSDEV-1105] Contribution. Allow commas in list name and update error message.

### PR DESCRIPTION
**Pull request by Innavin369**
[OSDEV-1105](https://opensupplyhub.atlassian.net/browse/OSDEV-1105) Contribution. Allow commas in list name and update error message.

* Updated error message
* Added tag parser for styling the errors

Screenshot 2024-06-26 at 13 16 12

### CODE REVIEW

File: src/react/src/components/ContributeForm.jsx [hunk 1]
```diff
@@ -114,11 +115,25 @@ class ContributeForm extends Component {
...
-                        {error.map(err => (
-                            <li key={err} style={{ color: 'red' }}>
-                                {err}
-                            </li>
-                        ))}
+                        {error.map(err => {
+                            const parsedString = parser
+                                .parseFromString(err, 'text/html')
+                                .querySelector('body').innerHTML;
+                            function createMarkup() {
```
Comment (VadimKovalenkoSNF) [inline src/react/src/components/ContributeForm.jsx:122]: 1. Can this be replaced into separate component?
2. This looks like anti-pattern, you try to get access to the DOM within render phase without using React refs.
3. You created `function createMarkup()` inside `.map` iteration which doesn't look clean in terms of performance.

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]

File: src/react/src/components/ContributeForm.jsx (hunk 1 above)
Comment (Innavin369) [reply src/react/src/components/ContributeForm.jsx]: rewritten

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Looks clearer now. I'd like to see unit tests in this PR (check `src\react\src\__tests__\utils.tests.js`).

File: doc/release/RELEASE-NOTES.md [hunk 2]
```diff
@@ -26,10 +26,12 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 * *Describe bugfix here.*
 
 ### What's new
+* [OSDEV-1105](https://opensupplyhub.atlassian.net/browse/OSDEV-1105) Contribution. Allow commas in list name and update error message.
 * [OSDEV-272](https://opensupplyhub.atlassian.net/browse/OSDEV-272) Facility Claims Page. Implement ascending/descending and alphabetic sort on FE.
 
 ### Release instructions:
-* *Provide release instructions here.*
+* Update code.
```
Comment (roman-stolar) [inline doc/release/RELEASE-NOTES.md:33]: I suppose this changes not needed.

File: doc/release/RELEASE-NOTES.md [hunk 3]
```diff
@@ -26,11 +26,13 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 * *Describe bugfix here.*
 
 ### What's new
+* [OSDEV-1105](https://opensupplyhub.atlassian.net/browse/OSDEV-1105) Contribution. Allow commas in list name and update error message.
 * [OSDEV-272](https://opensupplyhub.atlassian.net/browse/OSDEV-272) Facility Claims Page. Implement ascending/descending and alphabetic sort on FE.
 
 ### Release instructions:
 * *Provide release instructions here.*
 
+
```
Comment (roman-stolar) [inline doc/release/RELEASE-NOTES.md:35]: remove extra space

Comment (Innavin369) [reply doc/release/RELEASE-NOTES.md:35]: I guess we have to keep 2 empty lines after each release section.

Comment (vladsha-dev) [reply doc/release/RELEASE-NOTES.md:35]: Yes

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Review by VadimKovalenkoSNF — APPROVED** [review]

File: doc/release/RELEASE-NOTES.md (hunk 2 above)
Comment (Innavin369) [reply doc/release/RELEASE-NOTES.md]: replaced

**Pull request merged**